Github Issue Reader/Models/IssueModel.swift

```diff
@@ -6,17 +6,30 @@
 //
 
 import Foundation
+import UIKit
 
-struct Issue: Codable, Equatable {
+struct Issue: Codable, Hashable {
     let id: Int
-    let title: String?
+    let title: String
     let state: State
-    let user: User?
+    let user: User
     let body: String?
-    let createdAt: String?
-    
-    enum State: String, Codable, Equatable {
+    let createdAt: String
+
+    enum State: String, Codable, Hashable {
         case open
         case closed
     }
 }
+
+
+extension UIColor {
```

Comment:
Extensions aren't super complicated, you're adding method implementations to existing types. Sometimes you do it to types you don't own because you can't modify them directly. Sometimes you do it to your own stuff to organize code apart from one huge file. 

Sometimes you want to add functionality in certain circumstances. For example, you want Optional to be Equatable and Hashable when the thing it's holding is Equatable or Hashable. If this doesn't make sense we can write out this extension.

Reply:
I would like to do one with you. Your comment did help a little but there wasn't a "click" just yet.